vembrane/__init__.py

```diff
@@ -48,42 +52,51 @@
 }
 
 
-class Sample:
-    def __init__(self, record_idx: int, sample_name: str, sample: VariantRecordSample):
+class Format:
+    def __init__(
+        self, record_idx: int, name: str, record_samples: VariantRecordSamples
+    ):
         self._record_idx = record_idx
-        self._sample_name = sample_name
-        self._sample = sample
-        self._data = {}
+        self._name = name
+        self._record_samples = record_samples
+        self._sample_values = {}
 
-    def __getitem__(self, item):
+    def __getitem__(self, sample):
         try:
-            return self._data[item]
+            return self._sample_values[sample]
         except KeyError:
             try:
-                sample_format = self._sample[item]
-            except KeyError as ke:
-                raise UnknownFormatField(self._record_idx, self._sample_name, ke)
-            self._data[item] = sample_format
-            return sample_format
+                record_sample = self._record_samples[sample]
+            except KeyError:
+                raise UnknownSample(self._record_idx, sample)
+            value = record_sample[self._name]
```

Comment:
`record.sample[fmt]` is guaranteed to exist, right?
In the old code we had `{fmt: record.samples[sample][fmt] for fmt in record.format}`, hence I assume it is guaranteed, but just checking/asking :).

Reply:
I'm not too sure to be honest. I always assumed all samples need to have every FORMAT field specified in the header (the VCF specification isn't too clear on this).
So at this point in the code, if `record.format[fmt]` didn't already KeyError, `record.samples[sample][fmt]` shouldn't either.

> LGTM (barring a missing test case :P).
We should definitely add more testcases.